pmm-app/src/pmm-settings/components/PlatformLogin/SignUp.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { FC } from 'react';
+import { Field, Form, FormRenderProps } from 'react-final-form';
+import { LinkButton, useTheme } from '@grafana/ui';
+import validators from 'shared/components/helpers/validators';
+import { ButtonWithSpinner } from 'shared/components/Form';
+import { showErrorNotification, showSuccessNotification } from 'shared/components/helpers';
+import { Credentials, LoginFormProps } from './types';
+import { Messages } from './PlatformLogin.messages';
+import { CheckboxFieldAdapter, InputFieldAdapter } from './FieldAdapters/FieldAdapters';
+import { getStyles } from './PlatformLogin.styles';
+import { PlatformLoginService } from './PlatformLogin.service';
+import { PRIVACY_POLICY_URL, TERMS_OF_SERVICE_URL } from './PlatformLogin.constants';
+
+export const SignUp = ({ changeMode, getSettings }: LoginFormProps) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  const handleSignUpFormSubmit = async (credentials: Credentials) => {
+    try {
+      await PlatformLoginService.signUp(credentials);
+
+      getSettings();
+      showSuccessNotification({ message: Messages.signUpSucceeded });
+    } catch (e) {
+      console.error(e);
+      showErrorNotification({ message: Messages.errors.signUpFailed });
+    }
+  };
+
+  const CheckboxLabel = () => (
+    <span data-qa="sign-up-agreement-checkbox-label" className={styles.checkboxLabel}>
+      {`${Messages.agreementFirstPart} `}
+      <LinkButton className={styles.link} variant="link" href={TERMS_OF_SERVICE_URL}>
+        {Messages.termsOfService}
+      </LinkButton>
+      {` ${Messages.agreementSecondPart} `}
+      <LinkButton className={styles.link} variant="link" href={PRIVACY_POLICY_URL}>
+        {Messages.privacyPolicy}
+      </LinkButton>
+    </span>
+  );
```

Comment:
this could be brought outside with the checkbox field (creating a separate component), in order to simplify this code

Reply:
here i am afraid it will break the uniformity,  isn'it easier to read it when we have only `Field` components everywhere in this form?